### PR TITLE
Add more values to latest_coinmarketcap_data & project gql api

### PR DESCRIPTION
**Pull request by MilanRusev**
Added to projects api:
* Rank
* Volume in USD for the last 24h
* Available supply, total supply
* ROI = sum of all ROIs of all project's ICO-s. ROI = "price in usd now" / "price in usd at ico end"

### CODE REVIEW

**Comment by MilanRusev** [comment]
I flattened the latestCoinmarketcapData fields in the Project entity and deprecated the child entity latestCoinmarketcapData. The flattened fields are: rank, volumeUsd, availableSupply, totalSupply, priceUsd, marketcapUsd.

@Partysun I'll delete the latestCoinmarketcapData child entity when you change the client.

The new query should be:

```gql
query {
  allProjects
  {
    name,
    marketcapUsd,
    rank,
    volumeUsd,
    availableSupply,
    totalSupply,
    priceUsd
  }
}
```

instead of:

```gql
query {
  allProjects
  {
    name,
    latestCoinmarketcapData {
      rank,
      volumeUsd,
      availableSupply,
      totalSupply,
      priceUsd,
      marketCapUsd,
    }
  }
}
```

File: lib/sanbase/model/project.ex [hunk 1]
```diff
@@ -88,4 +89,43 @@ defmodule Sanbase.Model.Project do
...
+
+  def roi_usd(%Project{ticker: ticker, coinmarketcap_id: coinmarketcap_id} = project) when not is_nil(ticker) and not is_nil(coinmarketcap_id) do
+    with project <- Repo.preload(project, [:icos, :latest_coinmarketcap_data]),
+        true <- !is_nil(project.latest_coinmarketcap_data)
+              and !is_nil(project.latest_coinmarketcap_data.price_usd) do
+
+      zero = Decimal.new(0)
+
+      project.icos
+      |> Enum.reduce({project, 0, Decimal.new(0)}, &roi_usd_accumulator/2)
```
Comment (IvanIvanoff) [inline lib/sanbase/model/project.ex]: use `zero` here, too

File: lib/sanbase_web/graphql/schema/project_types.ex [hunk 2]
```diff
@@ -24,8 +24,11 @@ defmodule SanbaseWeb.Graphql.ProjectTypes do
     field :telegram_link, :string
     field :token_address, :string
     field :team_token_wallet, :string
-    field :market_cap_usd, :decimal
-    field :latest_coinmarketcap_data, :latest_coinmarketcap_data, resolve: assoc(:latest_coinmarketcap_data)
+    field :latest_coinmarketcap_data, :latest_coinmarketcap_data do
+      resolve assoc(:latest_coinmarketcap_data)
+
+      deprecate "The chiild entity latestCoinmarketcapData will be deleted. Please use the flattened fields."
```
Comment (IvanIvanoff) [inline lib/sanbase_web/graphql/schema/project_types.ex]: typo in `chiild`

File: lib/sanbase/prices/store.ex [hunk 3]
```diff
@@ -74,6 +74,14 @@ defmodule Sanbase.Prices.Store do
     |> parse_record
   end
 
+  def fetch_closest_price_point(pair, timestamp) do
```
Comment (IvanIvanoff) [inline lib/sanbase/prices/store.ex]: This is not the closest, but the closest **less** than the timestamp. Is it intentional?

Comment (MilanRusev) [reply lib/sanbase/prices/store.ex]: Yes, it should return the last price known for the given timestamp. The name is misleading.

Comment (valo) [reply lib/sanbase/prices/store.ex]: @MilanRusev can we improve the name, so that next person working on this understands it more easy?

Comment (valo) [reply lib/sanbase/prices/store.ex]: may be `fetch_last_price_point_before`?

Comment (MilanRusev) [reply lib/sanbase/prices/store.ex]: renamed it to `fetch_last_known_price_point`

File: lib/sanbase_web/graphql/resolvers/project_resolver.ex [hunk 4]
```diff
@@ -16,26 +17,32 @@ defmodule SanbaseWeb.Graphql.Resolvers.ProjectResolver do
 
   alias Sanbase.Repo
 
-  def all_projects(_parent, args, _context) do
+  def all_projects(_parent, args, context) do
```
Comment (IvanIvanoff) [inline lib/sanbase_web/graphql/resolvers/project_resolver.ex]: Not an error, but could lead to misunderstandings - the third argument is resolution (Absinthe.Resolution) and `context` is one subfield of that resolution

**Pull request merged**